mrrc/pkgs/indexing.py

```diff
@@ -100,3 +118,23 @@ def html_convert(tree: Tree, path: str, base_dir: str):
             index_file.write(index.generate_index_file_content())
 
     return html_files
+
+
+def load_exist_index(bucket: str, path: str) -> List[str]:
+    items = []
+
+    s3_client = S3Client()
+    try:
+        content = s3_client.read_file_content(bucket_name=bucket, key=path)
+        content = content.replace('\n', '').replace('\t', '')
+        print(content)
+    except ClientError as ex:
+        if ex.response['Error']['Code'] == 'NoSuchKey':
+            return []
+        else:
+            raise
+    root = fromstring(content)
+    for item in root.iter('a'):
+        items.append(item.attrib.get('title'))
```

Comment:
Seems this is using the previous index.html as a data source for the generation. Actually I don't think this is a good idea, because:
* We should always use "data" to generate "view", but should never use "view" as source of "data", because "view" is varying. You don't know when the view will be changed. For example, if our html template changed, like not use this a.title way but something else, your code will need to change too.( Remember the template is in a configurable location which means user can change it any time)
* If you really want to use this cache way, I'd recommend to add an extra file as a source of your cached data, like .index.txt or something similar. (Or maybe store them in object metadata? But need to think about the metadata length limit)